### PR TITLE
Suppress printing the internals of context.Context. Added a flag to o…

### PR DESCRIPTION
**Pull request by ebabani**
…verride this behaviour.

### CODE REVIEW

File: format/format.go [hunk 1]
```diff
@@ -124,6 +135,12 @@ func formatValue(value reflect.Value, indentation uint) string {
 		}
 	}
 
+	if !PrintContextObjects {
+		if value.Type().Implements(contextType) && indentation > 1 {
+			return "<suppressed>"
```
Comment (onsi) [inline format/format.go]: perhaps this should be <suppressed context>?  wdyt?

**Comment by onsi** [comment]
thanks @ebabani - i made a small comment on the commit.  can you also update the docs on gh-pages to mention this flag?  the relevant section is here:

http://onsi.github.io/gomega/#adjusting-output

**Comment by ebabani** [comment]
@onsi I forgot that context.Context was added in 1.7. Can probably create our own interface to keep backwards compatability.

**Comment by onsi** [comment]
ah good point.  yeah we'll need to do something to maintain backwards compatibility.... can you give it a shot?

File: format/format.go (hunk 1 above)
Comment (ebabani) [inline format/format.go]: What do you mean here?

Comment (onsi) [inline format/format.go]: oh weird, github sanitized my comment.. `"<suppressed context>"`

**Comment by ebabani** [comment]
Yeah, looks doable. The Context interface doesn't contain any 1.7 specific inputs or returns values.

**Pull request merged**